### PR TITLE
Add script to preprocess python documentation before publishing

### PR DESCRIPTION
**Pull request by xadupre**
**Description**:
Rename every folder starting with `_` in python documentation.

**Motivation and Context**
Github skips every folder starting with `_`.

### CODE REVIEW

File: tools/doc/rename_folders.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        renamed.append((r, name, into))
+        full_src = os.path.join(r, name)
+        full_into = os.path.join(r, into)
+        print("rename %r" % full_src)
+        os.rename(full_src, full_into)
+    
+    return renamed
+
+
+def replace_files(root, renamed):
```
Comment (natke) [inline tools/doc/rename_folders.py:32]: Can we document this function and explain why we need it?

File: docs/python/examples/plot_dl_keras.py [hunk 2]
```diff
@@ -85,7 +84,7 @@
 # Let's get more comprehensive results.
 
 if ok:
-    from keras.applications.densenet import decode_predictions
+    from tensorflow.keras.applications.densenet import decode_predictions
```
Comment (natke) [inline docs/python/examples/plot_dl_keras.py:87]: Minor nit pick but this does not seem to be related to the underscore changes ...

**Review by natke — COMMENTED** [review]
Can we add this to the CI?

File: docs/python/examples/plot_dl_keras.py (hunk 2 above)
Comment (xadupre) [reply docs/python/examples/plot_dl_keras.py:87]: No it is related to the documentation. I can remove it.

**Review by natke — APPROVED** [review]
